lib/livebook_web/live/session_live.ex

```diff
@@ -550,11 +555,13 @@ defmodule LivebookWeb.SessionLive do
 
   @impl true
   def handle_params(%{"cell_id" => cell_id}, _url, socket) do
+    assert_policy!(socket, :read)
```

Comment:
This handles `:cell_id` in routes relevant for editing (cell settings, upload, etc).